### DISCUSSION TITLE
Built a server on nas, access to port 8211 via ipv6 fails on the extranet, but the intranet can access the port

### DISCUSSION THREAD

**Discussion by BrynnChu** [thijsvanloef/palworld-server-docker#23] [Q&A]
Router has done port forwarding, tried turning off nas and router's firewall, still can't access it through the extranet，Anyone else having the same problem?

**Comment by conways-glider** [comment]
If this is an IPv6 issue, you should look into to your routers firewall settings. Most routers manage IPv6 via firewalls.

**Reply by BrynnChu** [reply]
Thank you for your reply. I have tried turning off the ipv6 firewall, tried using a reverse proxy, and I can never get access via ipv6.
But now I've got the public ipv4 address, and I can access the server port via ipv4 normally!

**Comment by YK-Samgo** [comment]
After hours of work, I figured out that the IPv6 support of this game is shit.

The server can only listen `0.0.0.0.:8211`, which is obviously IPv4 only.

I solved it with `socat -dd udp-listen:8212,reuseaddr,fork udp-connect:127.0.0.1:8211`. Now, in theroy, we can access the server by `ipv6.server:8212`. Note the `-dd` flag, this is the debug flag, and this will print the udp traffic through `socat`.

I tested it in game: 1. `ipv4.server:8211` worked; 2. `ipv4.server:8212` worked, and `socat` print the traffic; 3. `ipv6.server:8212` failed, and nothing happened within `socat`.

Conclusion: Palworld client didn't even send the request to ipv6 server! This is insane in 2024!

**Reply by sxyzy1016** [reply]
maybe bind the ipv6 address to a domain name?

**Reply by YK-Samgo** [reply]
I did, but nothing changed.

I add records to 2 domains:

domain1 IN A 192.168.1.XXX
domain1 IN AAAA 2409:XXX
domain2 IN AAAA 2409:XXX

When I tried to connect `domain1:8212`, the game connected `192.168.1.XXX:8212`.
When I tried to connect `domain2:8212`, the game just threw an error and `socat` reported nothing.

So I'm sure that this is a layer 2 (internet layer/ IP) issue. Maybe they use `AF_INET` instead of `AF_INET6`  when define socket object.

**Comment by TangJames** [comment]
What NAS are you using for this?

**Reply by BrynnChu** [reply]
synology 7.2.1

**Reply by TangJames** [reply]
Does it have enough RAM and CPU power for a palworld server?

**Reply by BrynnChu** [reply]
4core intel j4125  8g RAM ,
It's good enough for now, I'm thinking of adding an 8g memory to prevent memory overflow
